shell/src/main/java/org/apache/accumulo/shell/Shell.java

```diff
@@ -584,40 +586,39 @@ public int start() throws IOException {
 
         // If tab completion is true we need to reset
         if (tabCompletion) {
-          if (userCompletor != null) {
-            reader.removeCompleter(userCompletor);
-          }
-
           userCompletor = setupCompletion();
-          reader.addCompleter(userCompletor);
+          ((LineReaderImpl) reader).setCompleter(userCompletor);
         }
 
-        reader.setPrompt(getDefaultPrompt());
-        input = reader.readLine();
+        input = reader.readLine(getDefaultPrompt());
         if (input == null) {
-          reader.println();
```

Comment:
This println previously existed, so the user's bash shell or whatever, wouldn't print its prompt at the end of Accumulo's prompt, when existing by closing the input stream. I assume by removing this println, that no longer occurs? If it does, that should probably be put back.

Reply:
I don't believe this still occurs but I just noticed that CTRL-D will throw an `EndOfFileException` so I'll add a catch for that exception.

Reply:
> I don't believe this still occurs but I will add it back because I don't have a great way to test and a println doesn't hurt.

Manually/visual testing for this is sufficient.

> I also just noticed that CTRL-D will throw an `EndOfFileException` if I do it on a blank line so I'll add a catch for that exception. This might make that if statement redundant if that is the only situation input could be null.

Hmm, yeah, if it's causing that to explicitly be thrown, then it might be redundant. Handling it by catching the exception is much more clear. As long as the output doesn't get jumbled, and we get a new line when Ctrl-D is entered, I'm not too concerned about how the handling is implemented, as long as the behavior works.

Reply:
I will do more testing to confirm then push out my findings, CTRL-D for sure is hit by the exception and everything looks good there. I just need to make sure there is no other case input will equal null and to see what happens if something like that does happen.

Reply:
Another question I have, is if we want to exit on CTRL-D or for CTRL-C for that matter. I am in favor of not exiting on either of those two things but the current implementation is to exit the shell.

Reply:
From my testing, there is no reasonable way for a user to make input null beyond hitting CTRL-D, which sends a EOF exception. I interpret that as the user wanting to exit (which is standard for BASH but we can change that if we want). My latest commit reflects that idea.